lib/relationships/relationships.ts

```diff
@@ -0,0 +1,293 @@
+import * as _ from 'lodash';
+import { LinkConstraint } from '@balena/jellyfish-client-sdk/build/types';
+import {
+	constraints as sdkLinkConstraints,
+	getReverseConstraint,
+} from '@balena/jellyfish-client-sdk/build/link-constraints';
+
+/**
+ * This file contains the core relationships used by JF.
+ * It currently creates relationships based on link constraints and links created in different JF modules
+ * including:
+ * - jellyfis-client-sdk ( see relationshipsFromSdkLinkConstraints )
+ * - jellyfish-queue ( see relationshipsFromQueue )
+ * - jellyfish-worker ( see relationshipsFromWorker )
+ *
+ */
+
+interface Relationship {
```

Comment:
I think this should be defined and exported from autumndb as that's where the contract itself is defined.
Should also update it to follow the same pattern used for other contract type interfaces to keep things uniform.

Reply:
Right, I've refactored the type definitions and moved the type declarations to jellyfish-types

Reply:
@ramirogm We're actually looking to drop `jellyfish-types` as soon as possible and instead define TypeScript types next to the thing they define. `jellyfish-types` was useful when we moved the codebase from pure JavaScript to TypeScript, but now is just a painful relic that needs to be dropped. For example, it's quite painful to have to open two PRs in two places to add/update a contract - one for the contract itself and another for the TypeScript type.

Reply:
Moved the type definitions to https://github.com/product-os/jellyfish-types/pull/345

Reply:
@ramirogm I can help you find a place in autumndb because we really should avoid putting this in jellyfish-types. Putting it there will just create debt that we will need to pay back in the very near future. `org` etc type definitions will also be moved to the right places and out of jellyfish-types. I'll go ahead and archive jellyfish-types with a large notice at the top of the readme as its just causing unnecessary confusion.

Reply:
👍